library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -145,6 +146,33 @@ int calculateBatchStatus(long batchId) {
         return DownloadStatus.UNKNOWN_ERROR;
     }
 
+    boolean isBatchStartingForTheFirstTime(long batchId) {
+        Cursor cursor = null;
+        int hasStarted = 0;
+        try {
+            String[] projection = {DownloadContract.Batches.COLUMN_HAS_STARTED};
+            String[] selectionArgs = {String.valueOf(batchId)};
+
+            cursor = resolver.query(
+                    downloadsUriProvider.getBatchesUri(),
+                    projection,
+                    DownloadContract.Batches._ID + " = ?",
+                    selectionArgs,
+                    null
+            );
+
+            if (cursor.moveToFirst()) {
+                hasStarted = cursor.getInt(0);
+            }
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+
+        return hasStarted == DownloadContract.Batches.BATCH_HAS_STARTED ? false : true;
```

Comment:
Can simplify this to `return hasStarted != DownloadContract.Batches.BATCH_HAS_STARTED;`